File: testing_codes/Python/7-Zip_Bomb.py
import zipfile
import sys

def main():
    if len(sys.argv) < 2:
        print("Please specify a .zip file as an argument.", file=sys.stderr)
        return

    try:
        with zipfile.ZipFile(sys.argv[1], 'r') as z:
            for filename in z.namelist():
                try:
                    with z.open(filename) as file:
                        data = file.read()
                        # Perform some manipulation on the file data
                        # ...

                except Exception as e:
                    print(f"Failed to extract file: {filename}", file=sys.stderr)
    except Exception as e:
        print("Failed to open the .zip file.", file=sys.stderr)

if __name__ == '__main__':
    main()

File: testing_codes/Java/7-Zip_Bomb.java
import java.util.zip.*;
import java.io.*;

public class ZipReader {
    public static void main(String[] args) {
        if (args.length < 1) {
            System.err.println("Please specify a .zip file as an argument.");
            return;
        }

        try (ZipFile zipFile = new ZipFile(args[0])) {
            Enumeration<? extends ZipEntry> entries = zipFile.entries();
            while (entries.hasMoreElements()) {
                ZipEntry entry = entries.nextElement();
                if (entry.isDirectory()) continue;

                try (InputStream stream = zipFile.getInputStream(entry)) {
                    byte[] buffer = new byte[(int) entry.getSize()];
                    int readBytes = stream.read(buffer);

                    // Perform some manipulation on the file data
                    // ...

                } catch (IOException e) {
                    System.err.println("Failed to extract file: " + entry.getName());
                }
            }
        } catch (IOException e) {
            System.err.println("Failed to open the .zip file.");
        }
    }
}
